### PR TITLE
Add react@16.0.0-beta.2 w/ npm auto-update & update package.json

### LINKED ISSUE #11640: [Question] React 16 Alphas/Betas Not Available?
They've been published in NPM / unpkg and previous alphas are available; however, version 16 alphas do not show up in CDNJS.

Is something not working?

Reference: https://cdnjs.com/libraries/react

### PR DESCRIPTION
**Pull request by virgofx**
close #11640 
Update the repository's filename, homepage, npmName, npmFileMap, and author according to package.json of upstream GitHub repository and latest React 16 changes, close #11640

Pull request for issue: #11640
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/facebook/react
 * Official website (optional, not the repository): https://facebook.github.io/react/
 * NPM package url (optional): https://www.npmjs.com/package/react
 * License and its reference: https://github.com/facebook/react/blob/master/LICENSE
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 4,773
   - Count of stars: 72,332
   - Count of forks: 13,614
 * NPM download stats (optional):
   - Downloads in the last day: 181,649
   - Downloads in the last week: 1,181,361
   - Downloads in the last month: 4,710,677

# Essential checklist
 * [ ] I'm the author of this library
 * [X] I would like to add link to the page of this library on CDNJS on website / readme
 * [ ] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [X] The lib has notable popularity
   * [X] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [X] More than 500 downloads stats per month on npm registry
 * [X] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [X] Has valid tags for each versions (for git auto-update)
 * [X] Auto-update setup
 * [X] Auto-update target/source is valid.
 * [X] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [X] Separate unrelated changes into different commits.
 * [X] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [X] Close corresponding issue in commit message
 * [X] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/react/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "react",
-  "filename": "react.min.js",
-  "version": "15.6.1",
+  "filename": "umd/react.production.min.js",
+  "version": "16.0.0-beta.2",
```
Comment (PeterDaveHello) [inline ajax/libs/react/package.json]: Please don't bump version here.

File: ajax/libs/react/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "react",
-  "filename": "react.min.js",
-  "version": "15.6.1",
+  "filename": "umd/react.production.min.js",
```
Comment (PeterDaveHello) [inline ajax/libs/react/package.json]: Please leave the filename alone here as it's not changed in the latest stable version yet.

File: ajax/libs/react/package.json (hunk 1 above)
Comment (virgofx) [inline ajax/libs/react/package.json]: Will it pick up the latest 16 tree beta changes in directory structure in CI then?

File: ajax/libs/react/package.json [hunk 3]
```diff
@@ -13,19 +13,18 @@
...
-    ]
-  },
+  "homepage": "https://facebook.github.io/react/",
+  "npmName": "react",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "cjs/*,
+        "umd/*"
```
Comment (PeterDaveHello) [inline ajax/libs/react/package.json]: Shouldn't `cjs` and `umd` be the basePath?

Comment (virgofx) [inline ajax/libs/react/package.json]: Yes, but initial builds CI failed saying it couldn't find it. So was trying other approaches..

Directory structure is as follows:

```
.
├── cjs
│   ├── react.development.js
│   └── react.production.min.js
├── index.js
├── LICENSE
├── package.json
├── PATENTS
├── README.md
└── umd
    ├── react.development.js
    └── react.production.min.js
```

File: ajax/libs/react/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react",
-  "filename": "react.min.js",
+  "filename": "umd/react.production.min.js",
   "version": "15.6.1",
```
Comment (PeterDaveHello) [inline ajax/libs/react/package.json]: Please don't update filename here please.

File: ajax/libs/react/package.json (hunk 3 above)
Comment (PeterDaveHello) [inline ajax/libs/react/package.json]: It doesn't need to find it, you just update the meta data.

File: ajax/libs/react/package.json (hunk 4 above)
Comment (virgofx) [inline ajax/libs/react/package.json]: Version 16 is different...

**Comment by virgofx** [comment]
The tree structure is:
```
.
├── cjs
│   ├── react.development.js
│   └── react.production.min.js
├── index.js
├── LICENSE
├── package.json
├── PATENTS
├── README.md
└── umd
    ├── react.development.js
    └── react.production.min.js
```

Not sure best way to specify npmFileMap and the filename.

File: ajax/libs/react/package.json (hunk 4 above)
Comment (PeterDaveHello) [inline ajax/libs/react/package.json]: Yeah I know.

**Comment by PeterDaveHello** [comment]
hmmm ... looks like the files under `cjs` and `umd` sharing the same file name, okay, so maybe we don't need to put `cjs` and `umd` as basePath, but keep the directory structure.

**Comment by virgofx** [comment]
@sufuf3 Thanks for tips. Can you check now? Should be good

**Comment by sufuf3** [comment]
@virgofx Nice work.
May I ask that is `cjs/react.development.js` can work on web front-end?
If it can't use in front-end, maybe we have to remove this file in the first commit.

Thank you for your contribution.

**Comment by virgofx** [comment]
Both CJS and UMD can work on frontend. They provide additional development/debug warnings so they're useful. I would probably argue that the UMD one should be included as well "umd/*".

**Comment by sufuf3** [comment]
@virgofx Thank you for your explanation.
Then we have to modify from A to B in the first commit.
I really sorry give the wrong tip, because I didn't know `cjs/react.development.js` can use in front-end.
- A
```js
      "files": [
        "cjs/react.production*",
        "umd/*"
      ]
```
- B
```js
      "files": [
        "+(cjs|umd)/*"
      ]
```

Thank you.

**Comment by PeterDaveHello** [comment]
@sufuf3 Shouldn't we use "+(cjs|umd)/**/*"?

**Comment by virgofx** [comment]
It's more of a convention. Most packages use some sort of `/dist` or  `/build` output with only **one level** of files. What is the preferred CDNJS for package.json wildcard?

`+(cjs|umd)/**/*` - Current folder files + files from subfolders
`+(cjs|umd)/*` - Current folder files

**Comment by sufuf3** [comment]
@virgofx let's follow the maintainer said: use `"+(cjs|umd)/**/*"`
As you said incloude current folder files & files from subfolders is better.

And the change, could you modify in the first commit(`Add react@16.0.0-beta2 w/ npm auto-update` this one)?

It is helpful to us to trace git log.

Thank you very much.

**Comment by sufuf3** [comment]
And Please forgive me. Let's use `16.0.0-beta.2`in the first commit. 
I did not write well in my comment. 
CDNJS check every version of the library which is not in CDNJS. use `16.0.0-beta2` is not a good idea.

if it bothers you, I can help you.
I am sorry for any inconvenience this has caused you.
Thank you.

File: ajax/libs/react/package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "react",
-  "filename": "react.min.js",
-  "version": "15.6.1",
+  "filename": "umd/react.production.min.js",
+  "version": "16.0.0-beta2",
```
Comment (PeterDaveHello) [inline ajax/libs/react/package.json]: We don't use non-stable version as the default, so please rollback the version.

**Comment by virgofx** [comment]
@sufuf3 @PeterDaveHello Updated and also included separate PR #11647 for react-dom (The now distinct package beginning with v16)

**Comment by PeterDaveHello** [comment]
Ping @extend1994

**Comment by virgofx** [comment]
@extend1994 👍

**Comment by PeterDaveHello** [comment]
@virgofx thank you very much for your contribution!

**Pull request merged**